src/ztc/api/validators.py

```diff
@@ -194,3 +194,16 @@ def __call__(self, attrs):
                             f"Objects can't be updated with a relation to non-concept {field_name}"
                         )
                         raise ValidationError(msg, code=self.code)
+
+
+class ZaakInformatieObjectTypeCatalogusValidator:
+    code = "relations-incorrect-catalogus"
+    message = _("The zaaktype has catalogus different from informatieobjecttype")
+
+    def __call__(self, attrs: dict):
+        zaaktype = attrs.get("zaaktype")
+        informatieobjecttype = attrs.get("informatieobjecttype")
+
+        if zaaktype and informatieobjecttype:  # check in case of PATCH action
```

Comment:
If the zaaktype and informatieobjecttype are retrieved as I described above, this condition can be removed